AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/helper/APTCodeModelHelper.java

```diff
@@ -527,4 +530,28 @@ public TypeMirror getActualType(Element element, GeneratedClassHolder holder) {
 		TypeMirror type = actualTypes.get(element.asType().toString());
 		return type == null ? element.asType() : type;
 	}
+
+	public void addSuppressWarnings(JAnnotatable generatedElement, String annotationValue) {
+		Collection<JAnnotationUse> annotations = generatedElement.annotations();
+		for (JAnnotationUse annotationUse : annotations) {
+			if (annotationUse.getAnnotationClass().fullName().equals(SuppressWarnings.class.getCanonicalName())) {
+				JAnnotationValue value = annotationUse.getAnnotationMembers().values().iterator().next();
+				StringWriter code = new StringWriter();
+				JFormatter formatter = new JFormatter(code);
+				formatter.g(value);
+				if (!code.toString().contains(annotationValue)) {
```

Comment:
I changed this because now we do not need the `TypeElement` passed to this method, which is not always available. I know this is ugly and anti-pattern, but the other way is using 30 lines of reflective code to access the needed values from CodeModel. :worried: